doc/api/process.md

```diff
@@ -1538,19 +1542,21 @@ Will generate:
   rss: 4935680,
   heapTotal: 1826816,
   heapUsed: 650472,
-  external: 49879
+  external: 49879,
+  arrayBuffers: 9386
 }
 ```
 
-`heapTotal` and `heapUsed` refer to V8's memory usage.
-`external` refers to the memory usage of C++ objects bound to JavaScript
-objects managed by V8. `rss`, Resident Set Size, is the amount of space
-occupied in the main memory device (that is a subset of the total allocated
-memory) for the process, which includes the _heap_, _code segment_ and _stack_.
-
-The _heap_ is where objects, strings, and closures are stored. Variables are
-stored in the _stack_ and the actual JavaScript code resides in the
-_code segment_.
+* `heapTotal` and `heapUsed` refer to V8's memory usage.
+* `external` refers to the memory usage of C++ objects bound to JavaScript
+  objects managed by V8.
+* `rss`, Resident Set Size, is the amount of space occupied in the main
+  memory device (that is a subset of the total allocated memory) for the
+  process, including all C++ and JavaScript objects and code.
+* `arrayBuffers` refers to memory allocated for `ArrayBuffer`s and
+  `SharedArrayBuffer`s, including all Node.js [`Buffer`][]s.
+  This is also included in the `external` value. When Node.js is used as an
+  embedded library, this value may be `0`.
```

Comment:
sorry for being `raw` here, but I have these questions:

> When Node.js is used as an embedded library, this value may be `0`.

 - why? is it because Node.js cannot truly account for the allocations of the embedder?
 - where? which part of the code takes the decision based on the embedded state?
 - may be `0` - why again - why this decision cannot be deterministic as opposed to probabilistic? which factors decide the outcome? how would one use those factors to decide whether the value is trustworthy or not?

Reply:
> sorry for being `raw` here, but I have these questions:
> 
> > When Node.js is used as an embedded library, this value may be `0`.
> 
> * why? is it because Node.js cannot truly account for the allocations of the embedder?

Because embedders can specify their own `ArrayBuffer::Allocator`, which would not necessarily track allocations (e.g. `Electron` does this). 

> * where? which part of the code takes the decision based on the embedded state?

Inside the C++ `MemoryUsage()` binding, there’s a `array_buffer_allocator == nullptr` check. If there is no `NodeArrayBufferAllocator*` that the embedder informed us about, then the method sets the field for array buffers to 0.

> * may be `0` - why again - why this decision cannot be deterministic as opposed to probabilistic?

I don’t really understand this question, sorry – there is nothing probabilistic involved here, I think?

> which factors decide the outcome?

I think that’s also answered by the first question, but let me know if I’m misunderstanding.

> how would one use those factors to decide whether the value is trustworthy or not?

I think it’s safe to say that the value is trustworthy when it’s not `0`.

I didn’t want to go into a ton of detail in the documentation here – imo there should be an indication that the value may not *always* be provided, and that tools should not make the expectation that the value is non-zero. But for a tool it’s not really going to matter *why* exactly the value can’t be reported, I think, other than that it’s related to the fact that the Node.js code is running in a specific environment that behaves differently.

Reply:
> there is nothing probabilistic involved here, I think?

I termed it as probabilistic because of `may be 0` phrase. But based on your explanation to the rest of the questions, I understand it as: 

if embedders use only non-node.js allocators without a tracker, then this value `WILL` be 0 (not truly reflecting actual consumption), and if embedders rely on node.js' allocators, then the value will be accurate - zero or non-zero

Are we on the same page?

Reply:
@gireeshpunathil Yes, we are :+1: I feel like the most important thing is that the doc makes clear that this does not affect standard Node.js binaries.

Reply:
Maybe a simple "because such allocations may not be tracked" would be good?

Reply:
@jasnell Sure, done :+1: